File: app/src/main/java/br/com/vicentec12/mygames/ui/console/di/ConsoleComponent.java
package br.com.vicentec12.mygames.ui.console.di;

import br.com.vicentec12.mygames.ui.console.ConsoleActivity;
import dagger.Subcomponent;

@Subcomponent
public interface ConsoleComponent {

    @Subcomponent.Factory
    interface Factory {

        ConsoleComponent create();

    }

    void inject(ConsoleActivity mConsoleActivity);

}

File: app/src/main/java/br/com/vicentec12/mygames/ui/console/di/ConsoleComponent.kt
package br.com.vicentec12.mygames.ui.console.di

import br.com.vicentec12.mygames.di.ActivityScope
import br.com.vicentec12.mygames.ui.console.ConsoleActivity
import dagger.Subcomponent

@ActivityScope
@Subcomponent(modules = [ConsoleModule::class])
interface ConsoleComponent {

    @Subcomponent.Factory
    interface Factory {
        fun create(): ConsoleComponent
    }

    fun inject(mConsoleActivity: ConsoleActivity)

}